src/trexio_hdf5.c

```diff
@@ -406,43 +406,76 @@ trexio_exit_code trexio_hdf5_write_nucleus_num(const trexio_t* file, const uint6
   return TREXIO_SUCCESS;
 }
 
-trexio_exit_code trexio_hdf5_read_nucleus_coord(const trexio_t* file, double* coord) {
+trexio_exit_code trexio_hdf5_read_nucleus_coord(const trexio_t* file, double* coord, const int rank, const uint64_t* dims) {
 
   assert (file != NULL);
   assert (coord != NULL);
+  
+  trexio_hdf5_t* f = (trexio_hdf5_t*) file;
+  
+  herr_t status;
+  int rrank;
+  status = H5LTget_dataset_ndims (f->nucleus_group, NUCLEUS_COORD_NAME, 
+                                  &rrank);
+  assert (status >= 0);
+  assert (rrank == rank);
 
-  h5nucleus_t* nucleus = trexio_hdf5_read_nucleus((trexio_hdf5_t*) file);
+  hsize_t* ddims = (hsize_t*) calloc(rrank, sizeof(hsize_t));
```

Comment:
I would put `(int) rank` instead of `rrank` in this allocation, because you want the size of the table to be the size of what you give as an argument, not the size of what your read (although the values are equal).
I know it is the same for the computer, but I would find it easier to understand when you read the code.